src/braket/circuits/angled_gate.py

```diff
@@ -52,5 +53,10 @@ def angle(self) -> float:
         """
         return self._angle
 
+    def __eq__(self, other):
+        if isinstance(other, AngledGate):
+            return self.name == other.name and math.isclose(self.angle, other.angle)
```

Comment:
I was wondering if angles differing by 4pi should be considered equal as well, but that's probably a story for another day (and it complicates the stuff you can do here).

Reply:
Also, are we sure we want to treat close angles as equal? This does imply that equality is not transitive.

Reply:
I'm just paranoid about floating points; I do the same for matrices in #217 . Besides, the default tolerance of 1e09 should be more than enough.